### ISSUE TITLE
some benefits ofstarting mutations on char creation persist even if unselected (spells)

### ISSUE THREAD

**Issue by Ignaramico** [cataclysmbnteam/Cataclysm-BN#5693]
### Describe the bug

making a mod, noticed that if you select a mutation that gives spells in the third column, even after unselecting it and choosing another, you get both mutations spells,

> 

### Steps To Reproduce

make a mutations (at least 2) that gives spells that ends in the third column
make a profession that has one of them by default
at the trait selection window select the other one and unselect it
keep making the character as normal
at gameplay pop the spell window
and you end with both spells

### Screenshots

![Screenshot from 2024-11-09 12-34-02](https://github.com/user-attachments/assets/52e10de0-0d30-4055-ba4d-29c47dda72de)

### Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: bea1a3b [64-bit]
- Graphics Version: Tiles
- LAPI Version: 2
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [bn],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    No Rail Stations [No_Rail_Stations],
    Prevent Zombie Revivication [no_reviving_zombies],
    Limit Fungal Growth [limit_fungal_growth],
    Item Category Overhaul [Item_Category_Overhaul],
    A animist RP mod [Anmmod],
    No Fungal Monsters [No_Fungi]
]

### Additional context

here is the mutation syntax in case that the problem is me xD

`` {
    "type": "mutation",
    "id": "animist_necro_trait",
    "name": "animist:necromancer",
    "points": 0,
    "description": "At the start of the line to be a necromancer.",
    "starting_trait": true,
    "purifiable": false,
    "valid": false,
    "spells_learned": [ [ "summon_mon_skele", 1 ],[ "explode_summon_spell_necro", 1 ]  ]
    
  },
  {
    "type": "mutation",
    "id": "animist_spiritc_trait",
    "name": "animist:spirit caller",
    "points": 0,
    "description": "At the start of the line to be a spirit caller.",
    "starting_trait": true,
    "purifiable": false,
    "valid": false,
    "spells_learned": [ [ "guardian1_mark_spawn", 1 ],[ "explode_spirit_mark", 1 ] ]
    
  }```

**Comment by chaosvolt** [comment]
That is exactly why chargen-selectable spells in Arcana are bound to their associated mutation yeah, it's an old flaw that de-facto predates BN.

**Comment by Ignaramico** [comment]
going to check those then, to see the syntax parameters

**Comment by chaosvolt** [comment]
Basically the main thing it does is assign the associated mutation as its spell class, as seen here for example: https://github.com/chaosvolt/cdda-arcana-mod/blob/master/Arcana_BN/spells/spells_magic_signs.json#L29

This ensures that selecting and then de-selecting the offending mutation correctly forces chargen to remove that spell from the character's spell list. Far as I'm aware spells granted by professions at least don't seem to have this problem at least, just spells from mutations (if they do then that means it's gotten even worse and will also brick in-repo mod behavior, making it more urgent to fix).

**Comment by Ignaramico** [comment]
yeah i think the main problem is that my spells dont have a defined class, made an item too that gives a mutation when wield, and said mutation grants spells, you drop the item you lose the mutation but not the spells

**Comment by Ignaramico** [comment]
after further testing, even if the spells are classed, if you lose the mutation via another mutation (even if the second mutation has classed spells) that cancels it, you still retain the spells (from both mutations)
those test were made via item granting (using the relic data stuff) havent tested if it works the same if you use the mutate effect from spells
